texttunnel/processor.py

```diff
@@ -680,23 +686,18 @@ async def call_api(
                     response = await response.json()
 
                 if "error" in response:
-                    # Doesn't raise an exception, just adds the error to the result
+                    # API and rate limit errors don't raise an exception
+                    # They are found in the response JSON
                     logger.warning(
                         f"Request {self.task_id} failed with error {response['error']}"
                     )
 
-                    status_tracker.num_api_errors += 1
                     error = response
                     if "Rate limit" in response["error"].get("message", ""):
                         status_tracker.time_of_last_rate_limit_error = int(time.time())
                         status_tracker.num_rate_limit_errors += 1
-                        status_tracker.num_api_errors -= (
-                            1  # rate limit errors are counted separately
-                        )
-
-                    raise ValueError(
```

Comment:
Raising this error was pointless because the `if error` statement further below was going to run due to `error` not being `None` anyway